Reset Volume Manager during reload config & ca file rotation

chethanv28 opened the pull request:
<!-- Thanks for sending a pull request! -->

**What this PR does / why we need it**:
This PR is resetting VC instance in volume manager during reload config. Currently, the VC instance is reset only when VC hostname is changed. However with secure connection supported in WCP & Vanilla, the existing secure connections before reload config(before certificates regeneration) do not continue to work. Instead, the VC instance has to be invalidated(reset) during reload config.

Before the change:
```
2021-06-25T04:02:05.824Z	ERROR	vsphere/virtualcenter.go:380	failed to logout with err: Post "https://wdc-10-206-211-101.eng.vmware.com:443/sdk": x509: certificate signed by unknown authority	{"TraceId": "7424f9b9-7943-4115-94d1-e169eb5ecda3"}
...
...
2021-06-25T02:49:16.337Z	ERROR	wcp/controller.go:394	failed to create volume. Error: Post "https://wdc-10-206-211-101.eng.vmware.com:443/sdk": x509: certificate signed by unknown authority	{"TraceId": "83579658-378c-40c3-8b8f-78a0810da4ff"}
```

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Testing done**:
After the change:
```
2021-06-25T04:02:05.824Z	ERROR	vsphere/virtualcenter.go:380	failed to logout with err: Post "https://wdc-10-206-211-101.eng.vmware.com:443/sdk": x509: certificate signed by unknown authority
...
...
2021-06-25T04:08:08.945Z	INFO	volume/manager.go:373	CreateVolume: VolumeName: "pvc-4d686822-d5b4-4c1c-a482-4eb2ce806429", opId: "70232970"	{"TraceId": "885b6eda-de09-4220-aea5-05d57743a1be"}
2021-06-25T04:08:08.950Z	INFO	volume/util.go:324	Volume created successfully. VolumeName: "pvc-4d686822-d5b4-4c1c-a482-4eb2ce806429", volumeID: "1cecb4cc-2441-45ae-ab73-cf0c1757002f"	{"TraceId": "885b6eda-de09-4220-aea5-05d57743a1be"}
2021-06-25T04:08:08.950Z	DEBUG	volume/util.go:326	CreateVolume volumeId {{} "1cecb4cc-2441-45ae-ab73-cf0c1757002f"} is placed on datastore "ds:///vmfs/volumes/31851da9-724c4301/"	{"TraceId": "885b6eda-de09-4220-aea5-05d57743a1be"}
2021-06-25T04:08:08.955Z	DEBUG	wcp/controller.go:418	Volume Accessible Topology: []	{"TraceId": "885b6eda-de09-4220-aea5-05d57743a1be"}
2021-06-25T04:08:09.056Z	DEBUG	k8sorchestrator/k8sorchestrator.go:680	pvUpdated: PV pvc-4d686822-d5b4-4c1c-a482-4eb2ce806429 went to Bound phase	{"TraceId": "3997e870-15fb-49c0-bb92-9c6d70fd0401"}
2021-06-25T04:08:09.056Z	DEBUG	k8sorchestrator/k8sorchestrator.go:686	pvUpdated: Added '1cecb4cc-2441-45ae-ab73-cf0c1757002f -> storage-class-test-1/example-vanilla-block-pvc' pair to volumeIDToPvcMap	{"TraceId": "3997e870-15fb-49c0-bb92-9c6d70fd0401"}
2021-06-25T04:08:27.894Z	INFO	wcp/controller.go:556	DeleteVolume: called with args: {VolumeId:1cecb4cc-2441-45ae-ab73-cf0c1757002f Secrets:map[] XXX_NoUnkeyedLiteral:{} XXX_unrecognized:[] XXX_sizecache:0}	{"TraceId": "a0a7affd-5575-497c-bf8e-a26b873c1a0d"}
2021-06-25T04:08:27.894Z	DEBUG	common/vsphereutil.go:469	vSphere CSI driver is deleting volume: 1cecb4cc-2441-45ae-ab73-cf0c1757002f with deleteDisk flag: true	{"TraceId": "a0a7affd-5575-497c-bf8e-a26b873c1a0d"}
2021-06-25T04:08:27.902Z	INFO	vsphere/virtualcenter.go:269	vc.Config.CAFile /etc/vmware/wcp/tls/vmca.pem	{"TraceId": "a0a7affd-5575-497c-bf8e-a26b873c1a0d"}
2021-06-25T04:08:28.547Z	INFO	volume/manager.go:696	DeleteVolume: volumeID: "1cecb4cc-2441-45ae-ab73-cf0c1757002f", opId: "7023297e"	{"TraceId": "a0a7affd-5575-497c-bf8e-a26b873c1a0d"}
```

Steps performed for testing:
```
Step 1: Create volumes
Step 2: Performed CA file rotation
Step 3: Delete volumes succeeded
Step 4: Performed CA file rotation
Step 5: Created volume again & succeeded
```
Logs: https://gist.github.com/chethanv28/aa96a69c906627e6fb27bb4c6ee14b20

Running e2e pipelines

**Special notes for your reviewer**:

**Release note**:
<!--  Steps to write your release note:
1. Use the release-note-* labels to set the release note state (if you have access)
2. Enter your extended release note in the below block; leaving it blank means using the PR title as the release note. If no release note is required, just write `NONE`.
-->
```release-note
Reset Volume Manager during reload config & ca file rotation
```

svcbot-qecnsdp commented:
Started Vanilla block pre-checkin pipeline... Build Number: 111

svcbot-qecnsdp commented:
Started WCP block pre-checkin pipeline... Build Number: 74

lipingxue commented:
Change look good to me. Will approve it after the pipeline pass.

divyenpatel commented on pkg/common/cns-lib/volume/manager.go:
```diff
@@ -180,10 +180,8 @@ func (m *defaultManager) ResetManager(ctx context.Context, vcenter *cnsvsphere.V
-	}
+	log.Infof("Re-initializing defaultManager.virtualCenter")
+	managerInstance.virtualCenter = vcenter
 	m.virtualCenter.Config = vcenter.Config
```
if `managerInstance.virtualCenter = vcenter` is executed then it has the latest config. I think we do not need line 185?

chethanv28 replied:
Done

svcbot-qecnsdp commented:
``` 
Build ID: 74
WCP build status: SUCCESS 
Stage before exit: finally 
Jenkins E2E Test Results: 

Ran 14 of 196 Specs in 3021.172 seconds
SUCCESS! -- 14 Passed | 0 Failed | 0 Pending | 182 Skipped
PASS

Ginkgo ran 1 suite in 50m57.488806785s
Test Suite Passed
make: Leaving directory `/home/worker/workspace/csi-wcp-pre-check-in/Results/74/vsphere-csi-driver`

```

chethanv28 commented:
Performed the same steps for testing after addressing review comments:

```
2021-06-29T18:14:57.282Z	DEBUG	volume/util.go:326	CreateVolume volumeId {{} "1d763fe9-8c8f-48b2-a55e-a1b080ac78c7"} is placed on datastore "ds:///vmfs/volumes/bde1a16a-469933ac/"	{"TraceId": "41ce68d6-c75c-4195-a205-469f1227d5bf"}
2021-06-29T18:14:57.282Z	DEBUG	wcp/controller.go:418	Volume Accessible Topology: []	{"TraceId": "41ce68d6-c75c-4195-a205-469f1227d5bf"}
2021-06-29T18:14:57.319Z	DEBUG	k8sorchestrator/k8sorchestrator.go:680	pvUpdated: PV pvc-56fc4240-a0ff-4c48-bd54-14f9542c4985 went to Bound phase	{"TraceId": "0c91f3a9-eba0-4908-8090-381cda79d33b"}
2021-06-29T18:14:57.319Z	DEBUG	k8sorchestrator/k8sorchestrator.go:686	pvUpdated: Added '1d763fe9-8c8f-48b2-a55e-a1b080ac78c7 -> storage-class-test-1/example-vanilla-block-pvc' pair to volumeIDToPvcMap	{"TraceId": "0c91f3a9-eba0-4908-8090-381cda79d33b"}
2021-06-29T18:31:05.027Z	DEBUG	wcp/controller.go:150	fsnotify event: "\"/etc/vmware/wcp/tls/vmca.pem.tmp\": RENAME"	{"TraceId": "3904dd1c-4094-47be-a8e6-73514ad8b475"}
2021-06-29T18:31:05.135Z	DEBUG	wcp/controller.go:186	fsnotify event processed	{"TraceId": "3904dd1c-4094-47be-a8e6-73514ad8b475"}
2021-06-29T18:31:05.135Z	DEBUG	wcp/controller.go:144	Waiting for event on fsnotify watcher	{"TraceId": "3904dd1c-4094-47be-a8e6-73514ad8b475"}
2021-06-29T18:31:05.135Z	DEBUG	wcp/controller.go:150	fsnotify event: "\"/etc/vmware/wcp/tls/vmca.pem\": CREATE"	{"TraceId": "3904dd1c-4094-47be-a8e6-73514ad8b475"}
******************************CA FILE UPDATE************************
2021-06-29T18:31:05.135Z	INFO	wcp/controller.go:169	Observed ca file rotation at: "/etc/vmware/wcp/tls/vmca.pem"	{"TraceId": "3904dd1c-4094-47be-a8e6-73514ad8b475"}
2021-06-29T18:31:05.139Z	INFO	wcp/controller.go:227	Reloading Configuration	{"TraceId": "8ff9b36a-d690-4d62-bcfa-5a2cd82848f1"}
2021-06-29T18:31:05.140Z	DEBUG	config/config.go:382	GetCnsconfig called with cfgPath: /etc/vmware/wcp/vsphere-cloud-provider.conf	{"TraceId": "8ff9b36a-d690-4d62-bcfa-5a2cd82848f1"}
2021-06-29T18:31:05.155Z	DEBUG	config/config.go:282	Initializing vc server wdc-rdops-vm04-dhcp-58-72.eng.vmware.com	{"TraceId": "8ff9b36a-d690-4d62-bcfa-5a2cd82848f1"}
2021-06-29T18:31:05.155Z	INFO	config/config.go:321	No Net Permissions given in Config. Using default permissions.	{"TraceId": "8ff9b36a-d690-4d62-bcfa-5a2cd82848f1"}
2021-06-29T18:31:05.156Z	INFO	vsphere/utils.go:165	Defaulting timeout for vCenter Client to 5 minutes	{"TraceId": "8ff9b36a-d690-4d62-bcfa-5a2cd82848f1"}
2021-06-29T18:31:05.179Z	DEBUG	vsphere/virtualcenter.go:151	Setting vCenter soap client timeout to 5m0s	{"TraceId": "8ff9b36a-d690-4d62-bcfa-5a2cd82848f1"}
2021-06-29T18:31:05.419Z	INFO	vsphere/virtualcenter.go:177	New session ID for 'VSPHERE.LOCAL\workload_storage_management-d589def3-0bc6-4017-b4c6-d204c1a0d3b5' = 52a5bc5c-ff45-b226-3034-be0ce0159c68	{"TraceId": "8ff9b36a-d690-4d62-bcfa-5a2cd82848f1"}
2021-06-29T18:31:05.419Z	INFO	wcp/controller.go:254	Obtaining new vCenterInstance	{"TraceId": "8ff9b36a-d690-4d62-bcfa-5a2cd82848f1"}
2021-06-29T18:31:05.419Z	INFO	vsphere/virtualcenter.go:508	Initializing new vCenterInstance.	{"TraceId": "8ff9b36a-d690-4d62-bcfa-5a2cd82848f1"}
2021-06-29T18:31:05.420Z	INFO	vsphere/pbm.go:76	PbmClient wasn't connected, ignoring	{"TraceId": "8ff9b36a-d690-4d62-bcfa-5a2cd82848f1"}
2021-06-29T18:31:05.435Z	ERROR	vsphere/virtualcenter.go:376	failed to logout with err: Post "https://wdc-rdops-vm04-dhcp-58-72.eng.vmware.com:443/sdk": x509: certificate signed by unknown authority	{"TraceId": "8ff9b36a-d690-4d62-bcfa-5a2cd82848f1"}
2021-06-29T18:31:05.435Z	INFO	vsphere/virtualcentermanager.go:140	Successfully unregistered VC wdc-rdops-vm04-dhcp-58-72.eng.vmware.com	{"TraceId": "8ff9b36a-d690-4d62-bcfa-5a2cd82848f1"}
2021-06-29T18:31:05.435Z	INFO	vsphere/virtualcentermanager.go:118	Successfully registered VC "wdc-rdops-vm04-dhcp-58-72.eng.vmware.com"	{"TraceId": "8ff9b36a-d690-4d62-bcfa-5a2cd82848f1"}
2021-06-29T18:31:05.438Z	DEBUG	vsphere/virtualcenter.go:151	Setting vCenter soap client timeout to 5m0s	{"TraceId": "8ff9b36a-d690-4d62-bcfa-5a2cd82848f1"}
2021-06-29T18:31:05.641Z	INFO	vsphere/virtualcenter.go:177	New session ID for 'VSPHERE.LOCAL\workload_storage_management-d589def3-0bc6-4017-b4c6-d204c1a0d3b5' = 52c104e2-a6df-fdde-0ff5-3848b2632cc2	{"TraceId": "8ff9b36a-d690-4d62-bcfa-5a2cd82848f1"}
2021-06-29T18:31:05.641Z	INFO	vsphere/virtualcenter.go:542	vCenterInstance initialized	{"TraceId": "8ff9b36a-d690-4d62-bcfa-5a2cd82848f1"}
2021-06-29T18:31:05.642Z	INFO	wcp/controller.go:296	Successfully reloaded configuration	{"TraceId": "8ff9b36a-d690-4d62-bcfa-5a2cd82848f1"}
2021-06-29T18:31:05.642Z	INFO	wcp/controller.go:173	Successfully re-established connection with VC from: "/etc/vmware/wcp/tls/vmca.pem"	{"TraceId": "3904dd1c-4094-47be-a8e6-73514ad8b475"}
2021-06-29T18:31:05.642Z	DEBUG	wcp/controller.go:186	fsnotify event processed	{"TraceId": "3904dd1c-4094-47be-a8e6-73514ad8b475"}
...
...
2021-06-29T18:31:48.736Z	DEBUG	common/vsphereutil.go:469	vSphere CSI driver is deleting volume: 1d763fe9-8c8f-48b2-a55e-a1b080ac78c7 with deleteDisk flag: true	{"TraceId": "1e1bcf0c-a547-4588-87f1-5bd21e5b8158"}
2021-06-29T18:31:50.111Z	INFO	volume/manager.go:693	DeleteVolume: volumeID: "1d763fe9-8c8f-48b2-a55e-a1b080ac78c7", opId: "57f2128e"	{"TraceId": "1e1bcf0c-a547-4588-87f1-5bd21e5b8158"}
2021-06-29T18:31:50.111Z	INFO	volume/manager.go:705	DeleteVolume: Volume deleted successfully. volumeID: "1d763fe9-8c8f-48b2-a55e-a1b080ac78c7", opId: "57f2128e"	{"TraceId": "1e1bcf0c-a547-4588-87f1-5bd21e5b8158"}
2021-06-29T18:31:50.111Z	DEBUG	common/vsphereutil.go:475	Successfully deleted disk for volumeid: 1d763fe9-8c8f-48b2-a55e-a1b080ac78c7, deleteDisk flag: true	{"TraceId": "1e1bcf0c-a547-4588-87f1-5bd21e5b8158"}
2021-06-29T18:31:50.349Z	DEBUG	k8sorchestrator/k8sorchestrator.go:699	PV: pvc-56fc4240-a0ff-4c48-bd54-14f9542c4985 deleted. Removing entry from volumeIDToPvcMap	{"TraceId": "4c8f96c1-7e22-4c40-8c39-f8947db1ad29"}
2021-06-29T18:31:50.350Z	DEBUG	k8sorchestrator/k8sorchestrator.go:703	k8sorchestrator: Deleted key 1d763fe9-8c8f-48b2-a55e-a1b080ac78c7 from volumeIDToPvcMap	{"TraceId": "4c8f96c1-7e22-4c40-8c39-f8947db1ad29"}

```

Also performed password rotation:
```
2021-06-29T18:33:49.204Z	DEBUG	volume/util.go:326	CreateVolume volumeId {{} "491dbfb4-8921-46ed-963b-afc7b16b3bbc"} is placed on datastore "ds:///vmfs/volumes/bde1a16a-469933ac/"	{"TraceId": "6529ccb7-0bd6-4bd2-b59c-122f204bad64"}
2021-06-29T18:33:49.204Z	DEBUG	wcp/controller.go:418	Volume Accessible Topology: []	{"TraceId": "6529ccb7-0bd6-4bd2-b59c-122f204bad64"}
2021-06-29T18:33:49.316Z	DEBUG	k8sorchestrator/k8sorchestrator.go:680	pvUpdated: PV pvc-8ea0ca4a-263b-4ca7-98b6-c70a5fd07043 went to Bound phase	{"TraceId": "ba7ecf62-b771-4315-a325-69e7feb91e0c"}
2021-06-29T18:33:49.316Z	DEBUG	k8sorchestrator/k8sorchestrator.go:686	pvUpdated: Added '491dbfb4-8921-46ed-963b-afc7b16b3bbc -> storage-class-test-1/example-vanilla-block-pvc' pair to volumeIDToPvcMap	{"TraceId": "ba7ecf62-b771-4315-a325-69e7feb91e0c"}




2021-06-29T18:34:02.592Z	DEBUG	wcp/controller.go:150	fsnotify event: "\"/etc/vmware/wcp/..2021_06_29_18_34_02.766886471\": CREATE"	{"TraceId": "3904dd1c-4094-47be-a8e6-73514ad8b475"}
2021-06-29T18:34:02.592Z	DEBUG	wcp/controller.go:186	fsnotify event processed	{"TraceId": "3904dd1c-4094-47be-a8e6-73514ad8b475"}
2021-06-29T18:34:02.592Z	DEBUG	wcp/controller.go:144	Waiting for event on fsnotify watcher	{"TraceId": "3904dd1c-4094-47be-a8e6-73514ad8b475"}
2021-06-29T18:34:02.592Z	DEBUG	wcp/controller.go:150	fsnotify event: "\"/etc/vmware/wcp/..2021_06_29_18_34_02.766886471\": CHMOD"	{"TraceId": "3904dd1c-4094-47be-a8e6-73514ad8b475"}
2021-06-29T18:34:02.593Z	DEBUG	wcp/controller.go:186	fsnotify event processed	{"TraceId": "3904dd1c-4094-47be-a8e6-73514ad8b475"}
2021-06-29T18:34:02.593Z	DEBUG	wcp/controller.go:144	Waiting for event on fsnotify watcher	{"TraceId": "3904dd1c-4094-47be-a8e6-73514ad8b475"}
2021-06-29T18:34:02.593Z	DEBUG	wcp/controller.go:150	fsnotify event: "\"/etc/vmware/wcp/..data_tmp\": RENAME"	{"TraceId": "3904dd1c-4094-47be-a8e6-73514ad8b475"}
2021-06-29T18:34:02.593Z	DEBUG	wcp/controller.go:186	fsnotify event processed	{"TraceId": "3904dd1c-4094-47be-a8e6-73514ad8b475"}
2021-06-29T18:34:02.593Z	DEBUG	wcp/controller.go:144	Waiting for event on fsnotify watcher	{"TraceId": "3904dd1c-4094-47be-a8e6-73514ad8b475"}
2021-06-29T18:34:02.593Z	DEBUG	wcp/controller.go:150	fsnotify event: "\"/etc/vmware/wcp/..data\": CREATE"	{"TraceId": "3904dd1c-4094-47be-a8e6-73514ad8b475"}
2021-06-29T18:34:02.593Z	DEBUG	wcp/controller.go:186	fsnotify event processed	{"TraceId": "3904dd1c-4094-47be-a8e6-73514ad8b475"}
2021-06-29T18:34:02.593Z	DEBUG	wcp/controller.go:144	Waiting for event on fsnotify watcher	{"TraceId": "3904dd1c-4094-47be-a8e6-73514ad8b475"}
******************************PASSWORD ROTATED************************
2021-06-29T18:34:02.593Z	DEBUG	wcp/controller.go:150	fsnotify event: "\"/etc/vmware/wcp/..2021_06_29_18_13_46.387421920\": REMOVE"	{"TraceId": "3904dd1c-4094-47be-a8e6-73514ad8b475"}
2021-06-29T18:34:02.593Z	INFO	wcp/controller.go:227	Reloading Configuration	{"TraceId": "a76e0ff7-da03-407a-8cd5-9772d81acc79"}
2021-06-29T18:34:02.595Z	DEBUG	config/config.go:382	GetCnsconfig called with cfgPath: /etc/vmware/wcp/vsphere-cloud-provider.conf	{"TraceId": "a76e0ff7-da03-407a-8cd5-9772d81acc79"}
2021-06-29T18:34:02.601Z	DEBUG	config/config.go:282	Initializing vc server wdc-rdops-vm04-dhcp-58-72.eng.vmware.com	{"TraceId": "a76e0ff7-da03-407a-8cd5-9772d81acc79"}
2021-06-29T18:34:02.601Z	INFO	config/config.go:321	No Net Permissions given in Config. Using default permissions.	{"TraceId": "a76e0ff7-da03-407a-8cd5-9772d81acc79"}
2021-06-29T18:34:02.603Z	INFO	vsphere/utils.go:165	Defaulting timeout for vCenter Client to 5 minutes	{"TraceId": "a76e0ff7-da03-407a-8cd5-9772d81acc79"}
2021-06-29T18:34:02.617Z	DEBUG	vsphere/virtualcenter.go:151	Setting vCenter soap client timeout to 5m0s	{"TraceId": "a76e0ff7-da03-407a-8cd5-9772d81acc79"}
2021-06-29T18:34:02.787Z	INFO	vsphere/virtualcenter.go:177	New session ID for 'VSPHERE.LOCAL\workload_storage_management-d589def3-0bc6-4017-b4c6-d204c1a0d3b5' = 5212ab3f-ccbd-25f7-be86-d9f7f5933d2d	{"TraceId": "a76e0ff7-da03-407a-8cd5-9772d81acc79"}
2021-06-29T18:34:02.787Z	INFO	wcp/controller.go:254	Obtaining new vCenterInstance	{"TraceId": "a76e0ff7-da03-407a-8cd5-9772d81acc79"}
2021-06-29T18:34:02.787Z	INFO	vsphere/virtualcenter.go:508	Initializing new vCenterInstance.	{"TraceId": "a76e0ff7-da03-407a-8cd5-9772d81acc79"}
2021-06-29T18:34:02.788Z	INFO	vsphere/pbm.go:76	PbmClient wasn't connected, ignoring	{"TraceId": "a76e0ff7-da03-407a-8cd5-9772d81acc79"}
2021-06-29T18:34:02.797Z	INFO	vsphere/virtualcentermanager.go:140	Successfully unregistered VC wdc-rdops-vm04-dhcp-58-72.eng.vmware.com	{"TraceId": "a76e0ff7-da03-407a-8cd5-9772d81acc79"}
2021-06-29T18:34:02.797Z	INFO	vsphere/virtualcentermanager.go:118	Successfully registered VC "wdc-rdops-vm04-dhcp-58-72.eng.vmware.com"	{"TraceId": "a76e0ff7-da03-407a-8cd5-9772d81acc79"}
2021-06-29T18:34:02.802Z	DEBUG	vsphere/virtualcenter.go:151	Setting vCenter soap client timeout to 5m0s	{"TraceId": "a76e0ff7-da03-407a-8cd5-9772d81acc79"}
2021-06-29T18:34:02.904Z	INFO	vsphere/virtualcenter.go:177	New session ID for 'VSPHERE.LOCAL\workload_storage_management-d589def3-0bc6-4017-b4c6-d204c1a0d3b5' = 523033a6-5714-0081-71e6-96cae9c61611	{"TraceId": "a76e0ff7-da03-407a-8cd5-9772d81acc79"}
2021-06-29T18:34:02.905Z	INFO	vsphere/virtualcenter.go:542	vCenterInstance initialized	{"TraceId": "a76e0ff7-da03-407a-8cd5-9772d81acc79"}
2021-06-29T18:34:02.906Z	DEBUG	k8sorchestrator/k8sorchestrator.go:738	Could not find the improved-csi-idempotency feature state in ConfigMap csi-feature-states. Setting the feature state to false	{"TraceId": "a76e0ff7-da03-407a-8cd5-9772d81acc79"}
2021-06-29T18:34:02.906Z	INFO	volume/manager.go:183	Re-initializing defaultManager.virtualCenter	{"TraceId": "a76e0ff7-da03-407a-8cd5-9772d81acc79"}
2021-06-29T18:34:02.908Z	INFO	volume/manager.go:187	VC client timeout is set to 5m0s	{"TraceId": "a76e0ff7-da03-407a-8cd5-9772d81acc79"}
2021-06-29T18:34:02.908Z	INFO	volume/manager.go:189	Done resetting volume.defaultManager	{"TraceId": "a76e0ff7-da03-407a-8cd5-9772d81acc79"}
2021-06-29T18:34:02.908Z	DEBUG	k8sorchestrator/k8sorchestrator.go:738	Could not find the improved-csi-idempotency feature state in ConfigMap csi-feature-states. Setting the feature state to false	{"TraceId": "a76e0ff7-da03-407a-8cd5-9772d81acc79"}
2021-06-29T18:34:02.908Z	INFO	volume/manager.go:133	Retrieving existing defaultManager...	{"TraceId": "a76e0ff7-da03-407a-8cd5-9772d81acc79"}
2021-06-29T18:34:02.908Z	DEBUG	wcp/controller.go:294	Updated manager.CnsConfig	{"TraceId": "a76e0ff7-da03-407a-8cd5-9772d81acc79"}
2021-06-29T18:34:02.909Z	INFO	wcp/controller.go:296	Successfully reloaded configuration	{"TraceId": "a76e0ff7-da03-407a-8cd5-9772d81acc79"}
2021-06-29T18:34:02.909Z	INFO	wcp/controller.go:155	Successfully reloaded configuration from: "/etc/vmware/wcp/vsphere-cloud-provider.conf"	{"TraceId": "3904dd1c-4094-47be-a8e6-73514ad8b475"}
```

svcbot-qecnsdp commented:
Started vanilla Block pipeline... Build Number: 352

gohilankit commented:
Change looks good. Are there any e2e tests that cover password rotation scenario in WCP? If not, it will be good to manually execute it in your setup just for sanity.

/approve

chethanv28 commented:
> Change looks good. Are there any e2e tests that cover password rotation scenario in WCP? If not, it will be good to manually execute it in your setup just for sanity.
> 
> /approve

Yes, I ran it manually and captured the logs here: https://github.com/kubernetes-sigs/vsphere-csi-driver/pull/1043#issuecomment-870827763

divyenpatel reviewed (approved):
/approve
/lgtm

svcbot-qecnsdp commented:
``` 
Block vanilla build status: FAILURE 
Stage before exit: e2e-tests 
Jenkins E2E Test Results: 
Ran 49 of 196 Specs in 16091.344 seconds
FAIL! -- 39 Passed | 10 Failed | 0 Pending | 147 Skipped
--- FAIL: TestE2E (16091.46s)
FAIL

Ginkgo ran 1 suite in 4h29m10.162094976s
Test Suite Failed
make: Leaving directory `/home/worker/workspace/Block-Vanilla/Results/352/vsphere-csi-driver`

```
